src/utils/filesystem_path.cpp

```diff
@@ -141,7 +141,11 @@ class FilesystemPathImplWindows : public FilesystemPathImpl
 
 	virtual bool isFile() override
 	{
-		return !isDirectory();
+		WIN32_FIND_DATA ffd;
```

Comment:
Instead of `FindFirstFile()`, wouldn't it be easier and more straightforward to use just `GetFileAttributes()`?
```cpp
virtual bool isFile() override
{
	auto attrs = GetFileAttributes(_path.c_str());
	return (attrs != INVALID_FILE_ATTRIBUTES &&
		(attrs & FILE_ATTRIBUTE_DIRECTORY) == 0);
}
```
I am not a Windows user, but `GetFileAttributes()` kind of more resembles the POSIX `stat()` function.

Reply:
I agree with @s3rvac that `GetFileAttributes` makes more sense here than `FindFirstFile`.

Reply:
I have implemented it as suggested and did a forced update. At first I thought that the above would incorrectly return true on INVALID_FILE_ATTRIBUTES since it is defined as -1. However, I see that the logical && and == causes it to indeed function properly.